django/db/models/functions/comparison.py

```diff
@@ -74,6 +74,19 @@ def as_oracle(self, compiler, connection, **extra_context):
         return self.as_sql(compiler, connection, **extra_context)
 
 
+class Collate(Func):
+    function = 'COLLATE'
+    template = '%(expressions)s %(function)s %(collation)s'
+
+    def __init__(self, expression, collation):
+        self.collation = collation
```

Comment:
@felixxm having slept on it I wonder if ensuring `collation` is a valid identifiers could be a better solution here.

```suggestion
    function = 'COLLATE'
    template = '%(expressions)s %(function)s %(collation)s'
    # Inspired from https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
    collation_re = _lazy_re_compile(r'^[\w\-]+$')

    def __init__(self, expression, collation):
        if not self.collation_re.match(collation):
            raise ValueError(f'Invalid collation name {collation}')
        self.collation = collation
```

The pattern is a bit more too strict (e.g. it doesn't allow references to collations defined in other schemas `"schema"."collation"`) but it should at least address our immediate concerns of SQL injection vector?

Reply:
Good idea, I will implement it :+1: